src/osd/OSD.cc

```diff
@@ -470,7 +470,7 @@ void OSDService::shutdown()
     recovery_request_timer.shutdown();
   }
 
-  osdmap = OSDMapRef();
+  publish_map(OSDMapRef());
```

Comment:
who will be accessing `osdmap` at this point?

Reply:
Normall it's ok, but we can not insure that async msgr's works will stop before OSD::shutdown(), so maybe it is better to protect it here.

Reply:
but both `OSD::ms_dispatch()` and `OSD::ms_fast_dispatch()` do check if `is_stopping()` and `service.is_stopping()` respectively. could you be more specific?

Reply:
I ever talked about it with Haomai(i fw a email to you),  I find the async worker may keep processing event when osd shutdown.. My thought is keep the shard_ptr usage be thread safe is fine on 
 whatever circumstances..

Reply:
@Aran85 i agree that async messenger's worker thread might be still processing messages while OSD is shutting down as we shut down the messengers after done with all of these cleanups.

actually, i don't object to your change. i just need to understand the problem better so i am able to assess the severity of it.

anyway, this change LGTM.